gcloud-java-dns/src/main/java/com/google/gcloud/dns/ChangeRequestInfo.java

```diff
@@ -325,8 +336,8 @@ static ChangeRequestInfo fromPb(com.google.api.services.dns.model.Change pb) {
 
   @Override
   public boolean equals(Object other) {
-    return (other instanceof ChangeRequestInfo)
-        && toPb().equals(((ChangeRequestInfo) other).toPb());
+    return other != null && other.getClass().equals(ChangeRequestInfo.class)
+        && other instanceof ChangeRequestInfo && toPb().equals(((ChangeRequestInfo) other).toPb());
```

Comment:
No need for `other instanceof ChangeRequestInfo` as that will always be true if it got there.